app/services/arcgis_api/response_validation.rb

```diff
@@ -0,0 +1,28 @@
+module ArcgisApi
+  # Faraday middleware to raise exception when response status is 200 but with error in body
+  class ResponseValidation < Faraday::Middleware
```

Comment:
is there a way to keep the response logic closer to where the request is made?

Reply:
right, sorry, I understand how it works, but if this is specific to one API, could we pull it into the class that's making those requests rather than separating it across files?

Reply:
I think it's possible to have similar logic for retries when making the request, and I think it'd be better to move it there instead of having a separate class.

Reply:
@mitchellhenke not exactly, for http status code 200 actually invalid response, once the response is returned to caller, it's out of faraday middleware chain, and wont be able to utilize faraday-retry gem which works as a faraday middleware.